content/post/22-04-24-progress-report-33.md

```diff
@@ -0,0 +1,283 @@
+
+{{< imgw "images/post/pr_33/split" >}}
+
+{{< newsection >}}
+### New Salvage Maps
+*Contributed by mirrorcult and retequizzle*
+
+Several new salvage maps have been added for all of your lootin' and shootin' needs. Make sure you're on the lookout for some new hostile mobs in the area courtesy of EmoGarbage404.
+
+Here's a look at one of the new salvage maps:
+
+{{< imgw "images/post/pr_33/salvage" >}}
+
+{{< newsection >}}
+## Minor Changes
+Due to length constraints, not everything can get a dedicated section in the progress report and instead will be found listed here.
+
+### New Features
+
+- Several structures now rotate with the grid they're on. *Contributed by InquisitivePenguin*
+- Some clothing can now be torn into cloth, which can now be crafted into gauze. *Contributed by mirrorcult*
+- Certain station announcements now have unique colors. *Contributed by HoofedEar*
+- Trash bags now visually enlarge as you fill them. *Contributed by Macoron*
+- Added Uniform Printers, a machine for making jumpsuits. *Contributed by Rane*
+- You can now wear a guitar on your back. *Contributed by lapatison*
+- Botany plant sizes now scale with potency. *Contributed by mirrorcult*
+- Gas analyzers now work on pipes. *Contributed by Partmedia*
+- Added health analyzers. Doctor PDAs also function as a slower health analyzer. *Contributed by Fishfish458 and Rane*
+- Crafting a stackable item will attempt to merge it with the stack in your hand. *Contributed by ElectroSR*
+- Stack merging is now predicted, making it easier to pick up several individual items. *Contributed by ElectroSR*
+
+### Fixes, Tweaks, and Balancing
+
+- Flashlights now last 3.75x longer, flares now last 3x longer. *Contributed by Fishfish458*
+- Hardsuits no longer make you as slow and hardsuit helmets are no longer sufficient protection in a vacuum on their own. *Contributed by mirrorcult*
+- Crayons are now deleted when used up. *Contributed by actually-reb*
+- Xenos are now space-proof. *Contributed by DogZeroX*
+- Greatly reduced insulated gloves drop rate in maintenance, added budget gloves in its place. *Contributed by Emisse*
+- Several tweaks to shuttle movement to make them feel nicer to fly. *Contributed by Radrark*
+- Fixed some issues with audio playing twice or popups appearing twice. *Contributed by ElectroSR*
+- Fixed popups moving with the player. *Contributed by DrSmugleaf*
+- Fixed buckling on rotated grids. *Contributed by metalgearsloth*
+- Fixed flickering in the character editor. *Contributed by elthundercloud*
+- Fixed scrubbers all sharing the same list of gases to scrub. *Contributed by vulppine*
+- Fixed violating the laws of classical mechanics via pulling. *Contributed by metalgearsloth*
+
+{{< newsection >}}
+## Technical
+### Explosion Refactor
+*Contributed by ElectroSR*
+
+TODO: Either write or nuke this section
```

Comment:
Looking at other technical sections this might be overly long, in which case the third paragraph can probably just be omitted.

```suggestion
The old explosion code has been completely replaced. Previously explosions worked by using a simple raycast to check for obstructions, and then dealt damage to exposed entities based on their distance to the epicentre. Not only was this actually quite slow for large explosions, it also meant that any explosion would only ever destroy a single layer of walls.

The new explosion code uses a grid-based algorithm that iteratively looks for neighbouring tiles to add to the explosion. Airtight entities (i.e., walls) effectively delay the flood-fill in some directions, with a delay proportional to their health. The end result is that explosion can now properly penetrate multiple layers of walls, with each layer appropriately reducing the damage dealt to the next one. Additionally, the area that gets affected by an explosions is shaped by airtight obstacles. Explosions can now go around corners, and will deal more damage and travel further in confined areas.

The most convoluted part of the new explosion code is the need to support multiple rotated & translated shuttles and stations. It is challenging to have grid-based explosions that properly propagate onto and off of these grids, while also being blocked by airtight entities on those grids, and there are still some edge cases were explosions can skip past walls. Another challenge was ensuring that all the code is fast enough to allow for real-time nuking of the station. The biggest bottleneck is currently due to entity spawning and deletion, but there is also some physics slowdown due to having thousands of entities launched into space. Currently a station-destroying explosion still causes very noticeable stuttering, but there is still plenty of room for improvement.
```